corehq/apps/userreports/views.py

```diff
@@ -73,7 +73,7 @@
 def get_datasource_config_or_404(config_id, domain):
     try:
         return get_datasource_config(config_id, domain)
-    except DataSourceConfigurationNotFoundError:
+    except (DataSourceConfigurationNotFoundError, BadSpecError):
         raise Http404
```

Comment:
interesting. 

I think if we make `DataSourceConfigurationNotFoundError` a subclass of `BadSpecError` then remove that reraise code this works exactly how it should, no?